src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -762,17 +798,64 @@ export class SearchSource {
     return field;
   }
 
+  public async loadDataViewFields() {
+    const dataView = this.getField('index');
+    if (!dataView || !(dataView instanceof DataViewLazy)) {
+      return;
+    }
+    // eslint-disable-next-line no-console
+    console.log('loadDataViewFields because of DataViewLazy');
+    const request = this.mergeProps(this, { body: {} }, ['query', 'filter']);
+    let fields = dataView.timeFieldName ? [dataView.timeFieldName] : [];
+    const sort = this.getField('sort');
+    if (sort) {
+      const sortArr = Array.isArray(sort) ? sort : [sort];
+      for (const s of sortArr) {
+        const keys = Object.keys(s);
+        fields = fields.concat(keys);
+      }
+    }
+    for (const query of request.query) {
+      if (query.query) {
+        const nodes = fromKueryExpression(query.query);
+        const queryFields = getKueryFields([nodes]);
+        fields = fields.concat(queryFields);
+      }
+    }
+    const filters = request.filters;
+    if (filters) {
+      const filtersArr = Array.isArray(filters) ? filters : [filters];
+      for (const f of filtersArr) {
+        fields = fields.concat(f.meta.field);
+      }
+    }
+    fields = fields.filter((f) => Boolean(f));
+
+    if (dataView.getSourceFiltering() && dataView.getSourceFiltering().excludes.length) {
+      // if source filtering is enabled, we need to fetch all the fields
+      await dataView.getFields({ fieldName: ['*'] });
+    } else if (fields.length) {
+      const fieldSpec = await dataView.getFields({
+        fieldName: fields,
+        // fieldTypes: ['date', 'date_nanos'],
+      });
+      const spec = Object.values(fieldSpec.getFieldMap()).map((field) => field.spec);
+      dataView.setFields(spec);
+    }
+  }
+
```

Comment:
ok, this is new, and just needed for the alerting execution, where in this POC a DataViewLazy is being used. in this code we figure out when fields we need to load, that's the timefield, sort field, query fields, filter fields. once we know which fields we need, we get those, and they can build the query to ES